server/src/modules/api/servarr-api/helpers/sonarr.helper.ts

```diff
@@ -330,32 +330,33 @@ export class SonarrApi extends ServarrApi<{
       );
 
       // loop seasons
-      data.seasons = data.seasons.map((s) => {
+      for (const s of data.seasons) {
         if (type === 'all') {
           s.monitored = false;
         } else if (
           type === 'existing' ||
           (forceExisting && type === s.seasonNumber)
         ) {
           // existing episodes only, so don't unmonitor season
-          episodes.forEach((e) => {
-            if (e.seasonNumber === s.seasonNumber) {
-              this.UnmonitorDeleteEpisodes(
-                +seriesId,
-                e.seasonNumber,
-                [e.id],
-                false,
-              );
-            }
-          });
+          const episodesForSeason = episodes.filter(
+            (e) => e.seasonNumber === s.seasonNumber,
+          );
+
+          if (episodesForSeason.length > 0) {
+            await this.UnmonitorDeleteEpisodes(
+              +seriesId,
+              s.seasonNumber,
+              episodesForSeason.map((x) => x.id),
+              false,
+            );
+          }
```

Comment:
I've had this change sat around for a while so going to stuff it in here. This just adds a missing await to `UnmonitorDeleteEpisodes` and removes the unnecessary for loop.